### PR TITLE
moved-from container with unequal allocator will not be modified

### PR DESCRIPTION
**Pull request by madmongo1**
Currently proposed as 3 commits for discussion and analysis.

Will squash before final review.

### CODE REVIEW

**Comment by vinniefalco** [comment]
not much to talk about here, it looks correct

**Comment by madmongo1** [comment]
OK, will squash and force-push

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+Version XXX:
```
Comment (vinniefalco) [inline CHANGELOG.md]: why not just put 285 here

Comment (madmongo1) [reply CHANGELOG.md]: in case it doesn't make it into 285

Comment (vinniefalco) [reply CHANGELOG.md]: XXX is what you put on a barrel full of dynamite

**Pull request closed without merging**